src/js/App/GlobalFilter/constants.js

```diff
@@ -84,3 +88,27 @@ export const generateFilter = async () => {
 
     return [data, currToken];
 };
+
+export const flatTags = (filter, encode = false, format = false) => {
```

Comment:
This one seems expensive. Is it possible to add some memoize wrapper?

Reply:
Sure thing!

Reply:
The memoize resolver is a bit complex, but I think it'll help us when resolving multiple values. For instance on dashboard this function will be called multiple times.